src/main/java/com/mixpanel/android/mpmetrics/MixpanelPushNotification.java

```diff
@@ -57,22 +60,46 @@ public Notification createNotification(Intent inboundIntent) {
                 setWhen(now).
                 setContentTitle(data.title).
                 setContentText(data.message).
-                setTicker(data.message).
+                setTicker(data.ticker == null ? data.message : data.ticker).
                 setContentIntent(contentIntent);
 
+        maybeSetSubTitle(data);
         maybeSetNotificationBarIcon(data);
         maybeSetLargeIcon(data);
         maybeSetExpandableNotification(data);
         maybeSetCustomIconColor(data);
         maybeAddActionButtons(data);
-        maybeSetChannel();
+        maybeSetChannel(data);
+        maybeSetNotificationBadge(data);
+        maybeSetTag(data);
+        maybeSetGroupKey(data, contentIntent);
+        maybeSetTime(data);
+        maybeSetVisibility(data);
 
         final Notification n = buildNotification();
-        n.flags |= Notification.FLAG_AUTO_CANCEL;
+        if (!data.sticky) {
+            n.flags |= Notification.FLAG_AUTO_CANCEL;
+        }
 
         return n;
     }
 
+    protected void maybeSetGroupKey(NotificationData data, PendingIntent contentIntent) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && data.groupKey != null) {
+            builder.setGroup(data.groupKey);
+        }
+    }
+
+    protected void maybeSetTag(NotificationData data) {
```

Comment:
so this isn't a `maybeSetTag`, it always sets it. Can we just do this as a part of the base notification setup?

Reply:
actually this is weird, it's basically a leaky abstraction

Reply:
thought about it more, seems ok given the constraints

Reply:
oh but we should still just do this in `createNotification` instead of in this method

Reply:
if we are making it part of the default setup what should the default tag val be?